BedrockServer.cpp

```diff
@@ -2139,68 +2120,64 @@ void BedrockServer::handleSocket(Socket* s, bool isControl) {
         // are in a `shutting down` state, then we finish up and exit. In any other case, we just wait in `poll` again
         // until we get some data or a disconnection.
         int pollResult = 0;
-        struct pollfd pollStruct = { s->s, POLLIN, 0 };
+        struct pollfd pollStruct = { socket.s, POLLIN, 0 };
 
         // As long as `poll` returns 0 we've timed out, indicating that we're still waiting for something to happen. In
         // that case, we'll loop again *unless* we're shutting down.
         while (!(pollResult = poll(&pollStruct, 1, 1'000))) {
             if (_shutdownState != RUNNING) {
                 SINFO("Socket thread exiting because no data and shutting down.");
-                s->state = Socket::CLOSED;
-                ::shutdown(s->s, SHUT_RDWR);
+                socket.shutdown(Socket::CLOSED);
                 break;
             } 
         }
 
         // If the above loop didn't close the socket due to inactivity at shutdown, let's handle the activity.
-        if (s->state != STCPManager::Socket::CLOSED) {
+        if (socket.state != STCPManager::Socket::CLOSED) {
             if (pollResult < 0) {
                 // This is an exceptional case, we'll just kill the socket if this happens and let the client reconnect.
                 SINFO("Poll failed: " << strerror(errno));
-                s->state = Socket::CLOSED;
-                ::shutdown(s->s, SHUT_RDWR);
+                socket.shutdown(Socket::CLOSED);
             } else {
                 // We've either got new data, or an error on the socket. Let's determine which by trying to read.
-                if (!s->recv()) {
+                if (!socket.recv()) {
                     // If reading failed, then the socket was closed.
-                    s->state = Socket::CLOSED;
-                    ::shutdown(s->s, SHUT_RDWR);
+                    socket.shutdown(Socket::CLOSED);
```

Comment:
This is another example of only changing socket state inside the `Socket` class, and it looks cleaner here too.